reactor-netty5-core/src/main/java/reactor/netty5/BufferFlux.java

```diff
@@ -255,24 +255,28 @@ public final Flux<String> asString(Charset charset) {
 	@SuppressWarnings("rawtypes")
 	public final BufferMono aggregate() {
 		return Mono.defer(() -> {
-					CompositeBuffer output = CompositeBuffer.compose(alloc);
+					AtomicReference<Buffer> refOutput = new AtomicReference<>();
```

Comment:
do we care about allocation overhead? if so how about making a class like 

```java
static class DefferedBlablabla extends AtomicReference<Buffer> implements Supplier<Mono<...>> {

	public Mono<...> get() {
	     return return map(Buffer::send)
							.collectList()
							.doOnDiscard(Send.class, Send::close)
							.handle((list, sink) -> {
								Buffer output = null;
								if (!list.isEmpty()) {
									output = alloc.compose(list);
									set(output);
								}
								if (output != null && output.readableBytes() > 0) {
									sink.next(output);
								}
								else {
									sink.complete();
								}
							})
							.doFinally(signalType -> {
							    var instance =  get()
								if (instance != null) {
									instance.close();
								}
							});
	}

}

Reply:
then it can be used as 

```java
return Mono.defer(new DefferedBlablabla(sink))...
```

Reply:
@OlegDokuka Please check the new commit